### PR TITLE
A few ideas for the linting docs

### PR DESCRIPTION
**Pull request by amannn**
Thanks a lot for your initiative to set up this page! Looks really cool already, here are a few changes I'd propose. Let me know what you think! :)

**Proposed changes:**
 - Use inline text link in navigation docs instead of callout to link to linting. Callouts make the docs pages unfortunately a bit noisy to read, I'm trying to reduce them to really critical info :)
 - Mention "ESLint" on linting page, to be able to find it via search
 - Separate section for consistent usage of navigation APIs (so we could potentially add further sections in the future)
 - Use a snippet that works for either flat or legacy ESLint config, so that we could use tabs in the future for other linters like Biome
 - Showcase example to import from `@/navigation` (that's what we're proposing on the navigation page)
 - Remove usage of `next/router` as it's typically not used in App Router setups

### CODE REVIEW

File: docs/pages/docs/routing/navigation.mdx [hunk 1]
```diff
@@ -38,7 +33,7 @@ For example:
 
 ---
 
-Each strategy will provide you with corresponding [navigation APIs](#apis) that you'll typically provide in a central module to easily access them in components (e.g. `src/navigation.ts`).
+Each strategy will provide you with corresponding [navigation APIs](#apis) that you'll typically provide in a central module to easily access them in components (e.g. `src/navigation.ts`). To ensure cnosistent usage in your app, you can consider [linting for usage of these APIs](/docs/workflows/linting#consistent-usage-of-navigation-apis).
```
Comment (Gawdfrey) [inline docs/pages/docs/routing/navigation.mdx]:
```suggestion
Each strategy will provide you with corresponding [navigation APIs](#apis) that you'll typically provide in a central module to easily access them in components (e.g. `src/navigation.ts`). To ensure consistent usage in your app, you can consider [linting for usage of these APIs](/docs/workflows/linting#consistent-usage-of-navigation-apis).
```
Typo

File: docs/pages/docs/workflows/linting.mdx [hunk 2]
```diff
@@ -1,81 +1,29 @@
...
-      "error",
-      // Disallow the usage of the native Next.js `Link` component
-      {
-        "name": "next/link",
-        "message": "Please use <location of re-exported functions> instead."
-      },
-      // Disallow the specific functions that `next-intl` re-exports
+In this example, developers will be prompted to import from `@/navigation` when they try to import navigation APIs from Next.js.
+
+```javascript filename="eslint.config.js"
```
Comment (Gawdfrey) [inline docs/pages/docs/workflows/linting.mdx:11]: This is the flat config naming, but I guess that is the standard now so fine to just name it that way :)

**Review by Gawdfrey — APPROVED** [review]
I agree with all of the proposed changes!
LGTM

File: docs/pages/docs/routing/navigation.mdx (hunk 1 above)
Comment (amannn) [reply docs/pages/docs/routing/navigation.mdx]: Thanks!

File: docs/pages/docs/workflows/linting.mdx (hunk 2 above)
Comment (amannn) [reply docs/pages/docs/workflows/linting.mdx:11]: Yeah, I think for ESLint 8 it works too—I'd trust developers to do the right thing here :)

**Comment by Gawdfrey** [comment]
Just merge whenever you want

**Comment by amannn** [comment]
I think you need to merge here :). This is a PR against your fork.

**Pull request merged**